### PR TITLE
executor: remove Next function for IndexLookUpJoin

### PR DESCRIPTION
**Pull request by XuHuaiyu**
PTAL @coocood @zz-jason 
To #5985

### CODE REVIEW

File: executor/index_lookup_join.go [hunk 1]
```diff
@@ -177,40 +176,11 @@ func (e *IndexLookUpJoin) newInnerWorker(taskCh chan *lookUpJoinTask) *innerWork
...
-		return datumRow, nil
-	}
-}
-
-func (e *IndexLookUpJoin) prepareJoinResult(ctx context.Context, chk *chunk.Chunk, forChunk bool) error {
-	if !forChunk && e.joinResultCursor < chk.NumRows() {
-		return nil
-	}
-	e.joinResultCursor = 0
+func (e *IndexLookUpJoin) prepareJoinResult(ctx context.Context, chk *chunk.Chunk) error {
```
Comment (coocood) [inline executor/index_lookup_join.go]: Can we move the code into `NextChunk`?

File: executor/index_lookup_join.go [hunk 2]
```diff
@@ -212,6 +207,7 @@ func (e *IndexLookUpJoin) prepareJoinResult(ctx context.Context, chk *chunk.Chun
 			return nil
 		}
 	}
+	return nil
```
Comment (coocood) [inline executor/index_lookup_join.go]: We don't need to add this line.

Comment (XuHuaiyu) [reply executor/index_lookup_join.go]: @coocood Removed.

**Pull request merged**